cli/cache/http_cache/local.rs

```diff
@@ -49,36 +49,59 @@ impl LocalLspHttpCache {
   }
 
   pub fn get_file_url(&self, url: &Url) -> Option<Url> {
-    {
+    let sub_path = {
       let data = self.cache.manifest.data.read();
       if let Some(data) = data.get(url) {
-        if let Some(path) = &data.path {
-          return Url::from_file_path(self.cache.path.join(path)).ok();
-        }
+        url_to_local_sub_path(url, &data.content_type_header_map()).ok()?
+      } else {
+        url_to_local_sub_path(url, &Default::default()).ok()?
       }
-    }
-    match self.cache.get_cache_filepath(url, &Default::default()) {
-      Ok(path) if path.exists() => Url::from_file_path(path).ok(),
-      _ => None,
+    };
+    let path = sub_path.as_path_from_root(&self.cache.path);
+    if path.exists() {
+      Url::from_file_path(path).ok()
+    } else {
+      None
     }
   }
 
   pub fn get_remote_url(&self, path: &Path) -> Option<Url> {
     let Ok(path) = path.strip_prefix(&self.cache.path) else {
       return None; // not in this directory
     };
-    let has_hashed_component = path
+    let components = path
       .components()
-      .any(|p| p.as_os_str().to_string_lossy().starts_with('#'));
-    if has_hashed_component {
-      // check in the manifest
-      {
-        let data = self.cache.manifest.data.read();
-        if let Some(url) = data.get_reverse_mapping(path) {
-          return Some(url);
-        }
-      }
-      None
+      .map(|c| c.as_os_str().to_string_lossy())
+      .collect::<Vec<_>>();
+    if components
+      .last()
+      .map(|c| c.starts_with('#'))
+      .unwrap_or(false)
+    {
+      // the file itself will have an entry in the manifest
+      let data = self.cache.manifest.data.read();
+      data.get_reverse_mapping(path)
+    } else if let Some(last_index) =
+      components.iter().rposition(|c| c.starts_with('#'))
```

Comment:
I assume it's not possible to have a URL that has multiple hashes?

Reply:
There could be multiple hashed components. This selects the deepest hash in order to get the reverse mapping from there.